src/js/core/services/rowSearcher.js

```diff
@@ -221,6 +221,12 @@ module.service('rowSearcher', ['gridUtil', 'uiGridConstants', function (gridUtil
       }
     }
 
+    if (filter.flags.date === true) {
+      value = new Date(value);
+      // If the term has a dash in it, it comes through as '\-' -- we need to take out the '\'.
+      term = new Date(term.replace(/\\/g, ''));
```

Comment:
This filter is runnig ok. But will be better if we can use with formts like dd/mm/yyyy, mm/dd/yyy, etc..

In my case this filter obly works with date format yyyy-m-d (image). new Date parse this kind of format but not the others above.

![image](https://cloud.githubusercontent.com/assets/7308292/6725944/432bcae6-cdef-11e4-98c7-906fbde94bae.png)


Why not include moment.js in this project to support more formats ?

I think that we can use cellFilter (that is angular filter used to format the column date data) with moment.js ? 

I was playing with it and I think that could be a good solution.

This is my code :

```
if (column.cellFilter) {
	term = moment(term.replace(/\\/g, ''), column.cellFilter.split(':')[1].toUpperCase()).toDate();
} else {
	term = new Date(term.replace(/\\/g, ''));
}
```